File: python/125 - Valid Palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        regex = re.compile('[^a-zA-Z0-9]+')
        s = regex.sub('', s)
        s = s.lower()
        
        i = 0
        while i < len(s) / 2:
            if s[i] != s[len(s) - 1 - i]:
                return False
            i += 1
        return True

File: cpp/125 - Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        // var
        int left = 0;
        int right = s.length() - 1;

        while (left < right) {
            for (; left < right && !isalnum(s[left]); left++);
            for (; left < right && !isalnum(s[right]); right--);
            
            if (tolower(s[left]) != tolower(s[right]))
                return false;

            left++, right--;
        }
        return true;
    }
};